meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -58,42 +62,37 @@ if test "${fiovb.is_secondary_boot}" = "0"; then
 	# Only update bootcount when upgrade_available is set
 	if test "${fiovb.upgrade_available}" = "1"; then
 		setexpr bootcount ${fiovb.bootcount} + 1;
-		if test -n "${board_is_closed}"; then
+		if test -n "${fiovb_rpmb}"; then
 			fiovb write_pvalue bootcount ${bootcount};
+		else
+			# just save env with updated bootcount value
+			run saveenv_mmc
 		fi;
 
+
 		# If bootcount is above bootlimit, trigger rollback
 		# if not - try to boot secondary boot path for boot firmware validation
 		if test ${bootcount} -gt ${bootlimit}; then
-			echo "FIO: doing rollback..."
-			if test -n "${board_is_closed}"; then
-				fiovb write_pvalue rollback 1;
-				fiovb write_pvalue upgrade_available 0;
-				fiovb write_pvalue bootupgrade_available 0;
-			else
-				setenv rollback 1;
-				setenv upgrade_available 0;
-				setenv bootupgrade_available 0;
-			fi;
+			echo "${fio_msg} doing rollback..."
+
+			run rollback_setup
```

Comment:
Are we saving the environment here? I know a failed load_image will call rollback_setup + saveenv, but here it seems to be missing.

Reply:
yes, fixed that